src/com/palmergames/bukkit/towny/command/PlotCommand.java

```diff
@@ -698,14 +699,28 @@ public boolean parsePlotCommand(Player player, String[] split) throws TownyExcep
 								}
 								return true;
 							}
-						} 
-
-						WorldCoord worldCoord = new WorldCoord(world, Coord.parseCoord(player));
+						}
+						
+						try {
+							TownBlockType townBlockType = TownBlockType.lookup(split[0]);
 
-						setPlotType(resident, worldCoord, split[0]);
+							if (townBlockType == null)
+								throw new TownyException(TownySettings.getLangString("msg_err_not_block_type"));
+							
+							PlotPreChangeTypeEvent preEvent = new PlotPreChangeTypeEvent(townBlockType, townBlock);

```

Comment:
I missed it before but it's probably a good idea to pass along the resident or player who is changing the plot type, to make the event more usable. After that's done I think this will be ready to go.